tuna/miopen/subcmd/merge_db.py

```diff
@@ -254,16 +226,34 @@ def update_master_list(master_list, local_paths, mids, keep_keys):
           target_merge(master_list, key, vals, keep_keys)
 
 
+def is_float(num):
+  """Test if string can be interpreted as a float"""
+  try:
+    float(num)
+    return True
+  except ValueError:
+    return False
+
+
 def write_merge_results(master_list, final_file, copy_files):
   """write merge results to file"""
   # serialize the file out
   LOGGER.info('Begin writing to file: %s', final_file)
   with open(final_file, "w") as out_file:  # pylint: disable=unspecified-encoding
     for perfdb_key, solvers in sorted(master_list.items(),
                                       key=lambda kv: kv[0]):
+
       params = []
-      for solver_id, solver_params in sorted(
-          solvers.items(), key=lambda kv: (float(kv[1].split(',')[1]), kv[0])):
+      if is_float(list(solvers.values())[0].split(',')[0]):
```

Comment:
if not float/int then what could it be?

Reply:
There are 2 fdb signatures:
Old
```
992-7-7-1x1-128-7-7-8-0x0-1x1-1x1-0-NCHW-FP16-F=miopenConvolutionFwdAlgoImplicitGEMM:ConvMlirIgemmFwdXdlops,0.01904,0,miopenConvolutionFwdAlgoImplicitGEMM,<unused>;miopenConvolutionFwdAlgoDirect:ConvAsm1x1U,0.0528,0,miopenConvolutionFwdAlgoDirect,<unused>
```
New
```
992-7-7-1x1-128-7-7-8-0x0-1x1-1x1-0-NCHW-FP16-F=ConvMlirIgemmFwdXdlops:0.01904,0,miopenConvolutionFwdAlgoImplicitGEMM;ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm:0.024,0,miopenConvolutionFwdAlgoImplicitGEMM
```
Old way is keyed by algorithm and new way is keyed by solver. Notice that the fields have also changed and that kernel time is second in the old version, but first in the new version.